src/run/run_step.go

```diff
@@ -139,13 +139,22 @@ func run(ctx context.Context, state *core.BuildState, label core.BuildLabel, arg
 // environ returns an appropriate environment for a command.
 func environ(config *core.Configuration, setenv bool) []string {
 	env := os.Environ()
+	for _, e := range ad_run_environ() {
+		env = addEnv(env, e)
+	}
 	if setenv {
 		for _, e := range core.GeneralBuildEnvironment(config) {
 			env = addEnv(env, e)
 		}
 	}
 	return env
 }
+// ad_run_environ returns values that are appended to the environment for a command.
```

Comment:
probably just make this a top-level variable instead of a function.

also naming nit, should be camelCase for Go.